docs/guide/topics/threads.rst

```diff
@@ -0,0 +1,245 @@
+    async def secret_thread(ctx: commands.Context):
+        thread = await ctx.channel.create_thread(
+            name="Secret Birthday Planning!",
+            invitable=False
+        )
+        await thread.send(ctx.author.mention)
+
+We've chosen here to mention the author as soon as we make the thread.
+If we didn't do this, then the thread would be created, but we wouldn't necessarily be able to see it!
+
+Note that this mention is only guaranteed to work because we own the thread.
+Mentions from regular users wouldn't work in this thread, because it is not ``invitable``.
+
+You can also use the :meth:`~Thread.add_user` method to add someone, without sending a mention. This requires ``invitable`` to be ``False``.
+
+Unlike a public thread, private threads are not tied to messages, so they won't appear in the channel they're attached to.
+The member list will look like this inside in the thread:
+
+.. image:: /images/guide/threads/create_private_thread.png
+    :scale: 90
+
+
+.. warning::
+    Thread creation shows up in the audit logs, even if the thread is private.
+    Be wary of this if you don't intend for others to know about the thread.
+
+
+Adding members to threads
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We touched on it earlier a bit, but now we'll go into how to add people to threads.
+
+The go-to practice for doing this is pinging the member, like so:
+
+.. image:: /images/guide/threads/mention_member_to_thread.png
+
+The API has methods of adding members that do not require pinging them, which is implemented with :meth:`Thread.add_user`.
+You can add it as a command, like so:
+
+.. code-block:: python3
+    :emphasize-lines: 10
+
+    async def is_thread():
+        def predicate(ctx: commands.Context) -> bool:
+            return isinstance(ctx.channel, discord.Thread)
+        return commands.check(predicate)
+
+    @bot.command()
+    @is_thread() # we use the defined check above so that this command can only be used in a thread.
+    async def add_to_thread(ctx: commands.Context, *, member: discord.Member) -> None:
+        assert isinstance(ctx.channel, discord.Thread) # guarded by decorated check!
+        await ctx.channel.add_user(member)
+        await ctx.message.add_reaction("\U00002705")
+
+Which performs the following:
+
+.. image:: /images/guide/threads/add_member_to_thread.png
+
+Another method in which to do so is to mention a role within the thread.
+
+.. note::
+    There is a limit of around 100 members per addition to a thread, so if your role contains more than 100 members they will not be added.
+
+
+Setting the archive duration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Threads automatically archive after a set period of time of inactivity. You can choose this period when creating the thread.
+The current accepted values are:
+
+    - 1 hour
+    - 24 hours
+    - 3 days
+    - 1 week
+
+
+.. note::
+    The options for both "3 days" and "1 week" are locked behind server boost level 1 and 2, respectively.
```

Comment:
Possibly helpful to show how to write a check for if that feature is enabled (or rather if the guild fulfills the boost level)